experiments/base_experiment.py

```diff
@@ -510,13 +512,13 @@ def cycle(iterable):
                 )
 
         dt = time.time() - self.training_start_time
-        LOGGER.info(
```

Comment:
All this stuff can be done dynamically if we somehow extract len(self.train_loader), see comment above.